exchanges/orderbook/linked_list_test.go

```diff
@@ -460,11 +460,15 @@ func TestUpdateByID(t *testing.T) {
 		t.Fatalf("expecting %v but received %v", nil, err)
 	}
 
-	if a.retrieve()[1].Price == 0 {
+	if got := a.retrieve(2); len(got) != 2 && got[1].Price == 0 {
 		t.Fatal("price should not be replaced with zero")
 	}
 
-	if a.retrieve()[1].Amount != 1337 {
+	if got := a.retrieve(3); len(got) != 2 && got[1].Amount != 1337 {
```

Comment:
These `&&` tests don't work. You ask for 3 and check for len 2 and it doesn't fail

Reply:
🦅